sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeArrayWriter.java

```diff
@@ -47,126 +40,119 @@ private void assertIndexIsValid(int index) {
     assert index < numElements : "index (" + index + ") should < " + numElements;
   }
 
-  public void initialize(BufferHolder holder, int numElements, int elementSize) {
+  public UnsafeArrayWriter(UnsafeWriter writer) {
+    super(writer.getBufferHolder());
+  }
+
+  public void initialize(int numElements, int elementSize) {
```

Comment:
Should we move `elementSize` into the constructor? I don't think there are case where we are reusing `UnsafeArrayWriter `s.